components/renku_data_services/authz/authz.py

```diff
@@ -358,6 +358,37 @@ async def resources_with_permission(
                 ids.append(response.resource_object_id)
         return ids
 
+    async def resource_ids_for_user_membership(
```

Comment:
Why add this when `resources_with_permission` exists? (function just above)

You can see it used to filter namespaces by role here -> https://github.com/SwissDataScienceCenter/renku-data-services/blob/7ae9526c8f0ebcc78f23a2c390b6d07503d45053/components/renku_data_services/namespace/db.py#L298.

Reply:
The function is needed to properly get the projects where the user is a member with only Viewer permissions. If I use resources_with_permission with `Scope.WRITE`, it leaves out projects where the user is just a Viewer. But if I use `Scope.READ`, it brings in public projects the user can access, even though they’re not an owner or member.